lua/rustaceanvim/commands/diagnostic.lua

```diff
@@ -234,6 +240,106 @@ local function get_rendered_diagnostic(diagnostic)
   end
 end
 
+local function validate_window_dimensions(float_preview_lines, user_float_win_config)
+  local float_preview_config = vim.tbl_extend('keep', user_float_win_config, {
+    focus = false,
+    focusable = true,
+    focus_id = 'ra-render-diagnostic',
+    close_events = { 'CursorMoved', 'BufHidden', 'InsertCharPre' },
+  })
+
+  local c = vim.deepcopy(float_preview_config)
+  for _, key in ipairs { 'width', 'height', 'max_width', 'max_height' } do
+    c[key] = nil
+  end
+
+  local width, height = vim.lsp.util._make_floating_popup_size(float_preview_lines, c)
```

Comment:
issue: This is a private function, not part of Neovim's public API.

I think we can live without `validate_window_dimensions`. Most users won't set dimensions in their configs and by the looks of it, `vim.lsp.util.open_floating_preview` determines the correct size.

Although users setting boundaries in their configs could lead to the content not being able to fit, I don't think silently correcting `max_width` or `max_height` is desirable behaviour, as it could leave users confused about why their boundaries aren't being respected.